### PR TITLE
docs on distributed training

### PR DESCRIPTION
**Pull request by chrishkchris**
I have updated the documentation on distributed training

Since the NCCL comm API is not in version 2.0, I placed the docs on "Pre-release versions", i.e. http://localhost:3000/docs/next/dist-train:

Screenshot

### CODE REVIEW

File: docs-site/docs/assets/dist-train.md [hunk 1]
```diff
@@ -0,0 +1,420 @@
+---
```
Comment (FeynmanDNA) [inline docs-site/docs/assets/dist-train.md]: @chrishkchris why is this markdown doc in assets folder?

**Review by FeynmanDNA — CHANGES_REQUESTED** [review]

File: docs-site/docs/assets/dist-train.md (hunk 1 above)
Comment (chrishkchris) [reply docs-site/docs/assets/dist-train.md]: Sorry, I will delete this one (duplicated)

Comment (chrishkchris) [reply docs-site/docs/assets/dist-train.md]: Fixed

**Comment by chrishkchris** [comment]
Sorry, the duplicated file  is deleted

File: docs-site/docs/dist-train.md [hunk 2]
```diff
@@ -0,0 +1,432 @@
...
+
+&nbsp;
+
+3. Backward propagation and distributed parameter update, using half precision for gradient communication:
+
+```python
+sgd.backward_and_update_half(loss)
+```
+
+&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;It converts the gradients to 16 bits half precision format before allreduce
```
Comment (FeynmanDNA) [inline docs-site/docs/dist-train.md:52]: @chrishkchris Hi, sorry for troubling you again, but may I know if you need these `&nbsp;` in these markdown files? thanks

**Review by FeynmanDNA — CHANGES_REQUESTED** [review]

File: docs-site/docs/dist-train.md (hunk 2 above)
Comment (chrishkchris) [reply docs-site/docs/dist-train.md:52]: yes, those &nbsp create an empty space/line for better formatting when we view it on browser
thanks

Comment (FeynmanDNA) [reply docs-site/docs/dist-train.md:52]: got it. Thanks for the clarification!

File: docs-site/docs/assets/dist-train.md (hunk 1 above)
Comment (FeynmanDNA) [reply docs-site/docs/assets/dist-train.md]: ok!

**Review by FeynmanDNA — COMMENTED** [review]
fixed

**Pull request merged**